### PR TITLE
feat: update eslint to v8

### PR DESCRIPTION
**Pull request by ryuuji3**
Required to unblock using `prettier-eslint`

### CODE REVIEW

File: react/rules/jsx-control-statements.js [hunk 1]
```diff
@@ -1,38 +1,12 @@
...
-
+  "plugins": [
+    "jsx-control-statements"
+  ],
+  "extends": [
+    "plugin:jsx-control-statements/recommended"
+  ],
+  "rules": {
+    // Required for 7.0.0+ to avoid errors for If/Choose/etc.
+    "react/jsx-no-undef": ["error", { "allowGlobals": true }]
```
Comment (chris-skoblenick) [inline react/rules/jsx-control-statements.js:10]: Won't this turn off globals checking entirely? I believe there is a way to include our desired globals (the old `globals` block)?

File: react/rules/jsx-control-statements.md [hunk 2]
```diff
@@ -1,19 +1,5 @@
+## Chosen Rule Set
+For this languge, we have chosen the recommended rule set [].  If deviations from the rule set are made, they will be listed and discussed here.
```
Comment (chris-skoblenick) [inline react/rules/jsx-control-statements.md:2]: I think this meant to include a link to the recommended set documentation?

**Review by chris-skoblenick — CHANGES_REQUESTED** [review]

File: react/rules/jsx-control-statements.js (hunk 1 above)
Comment (ryuuji3) [reply react/rules/jsx-control-statements.js:10]: I looked through documentation and `allowGlobals` only "considers" items in the global scope.

File: react/rules/react.js [hunk 3]
```diff
@@ -6,65 +6,81 @@ const INDENTATION = 4;
...
+  "plugins": [
+    'react'
+  ],
 
-     // Code Smells - Warnings
-     "react/no-deprecated": "warn",
-     "react/no-find-dom-node": "warn",
+  // If these are not added, react/jsx-no-undef will find these are undefined.
+  // Ideally this block can be removed if a cleaner solution is found.
+  "globals": {
```
Comment (chris-skoblenick) [inline react/rules/react.js:16]: Shouldn't we have this in the `jsx-control-statements.js` config instead?

**Pull request merged**